java-backend/src/main/java/org/kframework/backend/java/symbolic/FastRuleMatcher.java

```diff
@@ -107,8 +106,17 @@ public FastRuleMatcher(GlobalContext global, int ruleCount) {
         ruleMask.stream().forEach(i -> constraints[i] = ConjunctiveFormula.of(context.global()));
         empty = BitSet.apply(ruleCount);
 
-        BitSet theMatchingRules = matchAndLog(subject.term(), pattern, ruleMask, List(), false);
+        if (global.javaExecutionOptions.logRulesPublic) {
+            System.err.format("\nRegular rule automaton phase, step %d\n" +
+                    "==========================================\n", step);
+        }
+        Term automatonLHS = global.getDefinition().automaton.leftHandSide();
```

Comment:
is there a specific reason for this change?  is it only for a better design or fixing an issue?

Reply:
i mean, for this change, we need to ensure that `definition` inside `fastComputeRewriteStep()` be the same with `global.getDefinition()` here, which may be true for now, but could be violated later.

Reply:
This was a parameter inline refactoring. I needed it to distinguish 3 phases of regular rule matching: automaton phase, attempting to match, and processing match results. And to print a message at the beginning of each phase.

Reply:
oh you mean that `global.getDefinition().automaton.leftHandSide()` involves some computation?

Reply:
ok, if you are convinced that "definition inside fastComputeRewriteStep() is the same with global.getDefinition() here", then i'm ok.

Reply:
Yes, it is the same. This was automatic refactoring.

Reply:
`global.getDefinition().automaton.leftHandSide()` does not involve compuation. But making this move helped put log messages closer to the context they document. Without this it was not clear how logs relate to the logic of functions.